react/react.d.ts

```diff
@@ -41,14 +41,8 @@ declare module React {
         propTypes?: ValidationMap<P>;
     }
 
-    interface ComponentClassType<P> extends ComponentStatics<P> {
-        new (): ComponentClass<P>;
-    }
-
-    interface ComponentClass<P> {
-        // Deprecated in 0.12. See http://fb.me/react-legacyfactory
-        // new(props: P): ReactElement<P>;
-        // (props: P): ReactElement<P>;
+    interface ComponentClass<P> extends ComponentStatics<P> {
+        new (): Component<P>;
```

Comment:
The correct type here is `new(props: P): ReactElement<P>` (see previously commented line). `Component<P>` instances are not directly constructable by users. I'd also appreciate it if you could leave the deprecation comment `// Deprecated in 0.12 ...etc`.